File: 2881-split-strings-by-separator/split-strings-by-separator.cpp
class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        vector<string>res;
        for (auto &word : words) {
            int start = 0;
            for (int i=0; i<=word.size(); i++) {
                if (i == word.size() || word[i] == separator) {
                    int length = i - start;

                    if (length > 0) {
                        res.push_back(word.substr(start, length));
                    }

                    start = i+1;

                }
            }
        }
        return res;
    }
};

File: 2881-split-strings-by-separator/split-strings-by-separator.js
/**
 * @param {string[]} words
 * @param {character} separator
 * @return {string[]}
 */
var splitWordsBySeparator = function(words, separator) {
    return words.flatMap(word => word.split(separator).filter(sep_word => sep_word.length > 0));
};